src/libponyc/ast/ast.c

```diff
@@ -1221,6 +1221,85 @@ void ast_free_unattached(ast_t* ast)
     ast_free(ast);
 }
 
+static void get_next_token_pos(ast_t* ast, size_t* line, size_t* pos)
+{
+  assert(ast != NULL);
+  
+  while (ast->child != NULL)
+  {
+    ast = ast->child;
+    while (ast->sibling != NULL)
+      ast = ast->sibling;
+  }
+
+  *line = token_line_number(ast->t);
+  *pos = token_line_position(ast->t);
+
+  token_id id = ast_id(ast);
+  if (id == TK_STRING || id == TK_ID)
+    *pos += token_string_len(ast->t);
```

Comment:
I'm trying to get the position one beyond the last visible character that the ast node contains.  I assume that if the deepest-rightmost node is not string or identifier, then it's zero-width, otherwise I add the string or identifier's width to the result.  I'm thinking that this is not a safe assumption for, e.g. int or float literals.

Is there a better way of getting the text spanned by an ast node?  Or should I just forget about quoting the parameter list in the docstring?